### PR TITLE
SOLR-16325 - JWT module Javadoc split package issue on JDK16+

### PR DESCRIPTION
**Pull request by risdenk**
https://issues.apache.org/jira/browse/SOLR-16325

### CODE REVIEW

**Comment by risdenk** [comment]
I'm still not sure I like this fix since we still have split packages in jwt-auth compared to core. This should probably be cleaned up further. I'm checking w/ @gerlowskija since this was introduced trying to get to v2 apis it looks like.

**Comment by gerlowskija** [comment]
LGTM - I went ahead and renamed the packages where the API classes live in `jwt-auth`.  They're now under `o.a.s.security.jwt.api`, which we probably should've done for the start.  This fixes the underlying split-package issue as well as fixing precommit/check.

I'm running tests right now but will probably merge this as soon as those pass in order to get the build fixed for folks.  Sorry I'd broken it for so long on these newer JDKs!

**Review by gerlowskija — APPROVED** [review]
LGTM, thanks Kevin!

File: solr/modules/jwt-auth/src/test/org/apache/solr/handler/admin/api/V2JWTSecurityApiMappingTest.java [hunk 1]
```diff
@@ -22,6 +22,7 @@
 import org.apache.solr.api.AnnotatedApi;
 import org.apache.solr.api.Api;
 import org.apache.solr.api.ApiBag;
+import org.apache.solr.security.jwt.api.ModifyJWTAuthPluginConfigAPI;
```
Comment (risdenk) [inline solr/modules/jwt-auth/src/test/org/apache/solr/handler/admin/api/V2JWTSecurityApiMappingTest.java:25]: I think the package on this test needs to be updated as well.

**Comment by madrob** [comment]
Do we need to update ref guide pages on configuring anything to reflect the new packages?

**Comment by risdenk** [comment]
Based on the change that introduced these files - https://github.com/apache/solr/pull/897/files - it doesn't look like any ref guide pages need to be updated.

**Comment by risdenk** [comment]
```
./gradlew check
...
BUILD SUCCESSFUL in 24m 33s
544 actionable tasks: 249 executed, 295 up-to-date
```

@gerlowskija looks good to me

**Review by janhoy — APPROVED** [review]

**Comment by gerlowskija** [comment]
Thanks for merging this Kevin - I've been trying to get a clean build all day; lots of unrelated fails.

**Pull request merged**